calliope/backend/pyomo/constraints/capacity.py

```diff
@@ -155,6 +163,36 @@ def energy_capacity_storage_constraint_rule(backend_model, loc_tech):
     )
 
 
+def energy_capacity_storage_constraint_rule(backend_model, loc_tech):
```

Comment:
Have you tried sending this off to `get_capacity_constraint`? I.e. `get_capacity_constraint(backend_model, 'energy_cap', loc_tech, _equals=energy_cap_per_storage_cap_equals, _max=energy_cap_per_storage_cap_max, _min=0 scale=None)`. I don't actually know if pyomo will complain that there is a decision variable in the upper bound. If it does, then I'm fine with this approach, but it misses a check in `get_capacity_constraint` as to whether the equals is infinite.

But now I think about this, the check would be much easier to put into preprocessing anyway (any variable with `_equals` in the name cannot have inf in its values)...

Reply:
I did try based on your recommendation, but this line did not work: `decision_variable = getattr(backend_model, parameter)`.

Reply:
ok, don't worry about it then!